server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java

```diff
@@ -31,13 +35,18 @@
 import org.apache.james.mailrepository.api.MailRepositoryStore;
 import org.apache.james.mailrepository.api.MailRepositoryUrl;
 import org.apache.james.server.core.MimeMessageInputStream;
+import org.apache.james.task.Task;
 import org.apache.james.util.streams.Iterators;
+import org.apache.james.vault.memory.DeleteByQueryExecutor;
```

Comment:
maybe `DeleteByQueryExecutor` should be located out side of `memory` package. I see it is used in mail repository vault

Reply:
Maybe I need to include it in another package